script.js

```diff
@@ -1,4 +1,128 @@
+//define vars
+var indivisualDiceRoll;
+var mode = `instructions`;
+var playerNumber = 1;
+var allValues = [];
+var numberOfPlayer = 2;
+var diceNumber = 2;
+var whoPlays = `player`;
+var winCondition = `highest value wins`;
+var diceNumber = 2;
+var allDiceRoll = [];
+var currentDice = 1;
+var category;
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (input == `instructions`) {
+    return instruction();
+  }
+  // if (input == `computer` || input == `player`) {
+  //   whoPlays = input;
+  //   return `Your player has been set to ${input}.Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+
+  if (input == `highest value wins` || input == `lowest value wins`) {
+    winCondition = input;
+    return `Your winning criteria has been set to ${input}.Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+  }
+  if (mode == `instructions`) {
+    mode = `roll dice`;
+    return instruction();
+  }
+  if (input == `player number` || input == `dice number`) {
+    mode = input;
+
+    return `Your mode has been set to ${input}. Please input the ${input} you want`;
+  }
+  if (mode == `player number`) {
+    numberOfPlayer = input;
+    mode = `instructions`;
+    return `The number of players have been set to ${numberOfPlayer}. Please press submit to start the game. `;
+  }
+  if (mode == `dice number`) {
+    diceNumber = input;
+    mode = `instructions`;
+    return `The number of dice have been set to ${diceNumber}. Please press submit.`;
+  }
```

Comment:
Up till this point, the main function seems very neat and clean. However, from this point on, there is alittle too much logic within the main function itself. Try to abstract out the logic into helper functions